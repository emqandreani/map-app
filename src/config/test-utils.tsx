import { StyleSystemProvider } from "@architecture-it/stylesystem";
import { configureStore, PreloadedState } from "@reduxjs/toolkit";
import { render, renderHook } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import getTheme from "@architecture-it/stylesystem/Theme";
import React, { PropsWithChildren } from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";

import { ModuleContext } from "../store/hooks";
import { RootState, store } from "../store/store";
import { reducers } from "../store/reducers";
// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: typeof store;
  basename?: string;
}

interface WithProvidersProps {
  basename?: string;
  store: typeof store;
}

const colors = {
  primary: "#d71920",
  colorText: "#616161",
  error: "#d71920",
  secondary: "#b6040b",
};

export function WithProviders({ basename, store }: WithProvidersProps) {
  return function Wrapper({ children }: PropsWithChildren<{}>) {
    return (
      <BrowserRouter basename={basename}>
        <StyleSystemProvider theme={getTheme(false, colors)}>
          <Provider context={ModuleContext} store={store}>
            {children}
          </Provider>
        </StyleSystemProvider>
      </BrowserRouter>
    );
  };
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: reducers, preloadedState }),
    // Basename might be "/" for modules
    basename = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: WithProviders({ basename, store }), ...renderOptions }),
  };
}

export function renderHookWithProviders<Result, Props>(
  hook: (_initialProps: Props) => Result,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: reducers, preloadedState }),
    // Basename might be "/" for modules
    basename = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...renderHook(hook, { wrapper: WithProviders({ basename, store }), ...renderOptions }),
  };
}
