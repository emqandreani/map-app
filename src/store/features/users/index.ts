import { createSlice } from "@reduxjs/toolkit";
import {
  type IState,
  fullfiledSimpleCallbackCase,
  pendingSimpleCallbackCase,
  rejectCallbackCase,
} from "@architecture-it/core";

import { IUsers } from "../../../domain/IUsers";
import { RootState } from "../../store";

import { getUsers } from "./asyncActions";

const defaultData: IUsers[] = [
  {
    id: 1,
    name: "Lionel",
    lastName: "Messi",
    age: 36,
  },
  {
    id: 2,
    name: "Juli√°n",
    lastName: "Alvarez",
    age: 23,
  },
  {
    id: 3,
    name: "Emiliano",
    lastName: "Martinez",
    age: 31,
  },
];

interface IUsersState extends IState<IUsers[]> {}

const initialState: IUsersState = {
  data: defaultData,
  error: null,
  isLoading: false,
};

export const usersSlice = createSlice({
  name: "contributor",
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    // Get Users
    builder.addCase(getUsers.pending, (state) => {
      pendingSimpleCallbackCase(state);
    });
    builder.addCase(getUsers.rejected, (state, action) => {
      rejectCallbackCase(state, action);
    });
    builder.addCase(getUsers.fulfilled, (state, _action) => {
      fullfiledSimpleCallbackCase(state);
      // Set the response API data to the store
      // state.data = action.payload;
    });
  },
});

export default usersSlice.reducer;

export const selectUsers = (store: RootState) => store.usersReducer;
