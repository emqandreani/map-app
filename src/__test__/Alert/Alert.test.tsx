import { fireEvent, waitFor } from "@testing-library/react";
import { CheckCircle, CircleXmark } from "@architecture-it/stylesystem";

import TemplateAlert, { getIcon } from "../../components/Alert";
import { renderWithProviders } from "../../config/test-utils";
import { IAlert } from "../../store/features/alert";

const defaultValues: IAlert = {
  message: "Colaborador creado con Ã©xito!",
  type: "success",
  open: true,
};

describe("<Alert />", () => {
  it("Should render without crash", async () => {
    const { getAllByTestId } = renderWithProviders(<TemplateAlert />, {
      preloadedState: { alertReducer: defaultValues },
    });

    const alerts = getAllByTestId("alert");

    await waitFor(() => {
      expect(alerts[0]).toMatchSnapshot();
    });
  });

  it("Should render the correct message", () => {
    const { getAllByText } = renderWithProviders(<TemplateAlert />, {
      preloadedState: { alertReducer: defaultValues },
    });

    const messages = getAllByText(defaultValues.message);

    expect(messages[0].innerHTML).toBe(defaultValues.message);
  });

  it("Should close the alert when triggering the handleOnClose() function", () => {
    const { getAllByTitle, store } = renderWithProviders(<TemplateAlert />, {
      preloadedState: { alertReducer: defaultValues },
    });

    const closeButtons = getAllByTitle("Close");

    fireEvent.click(closeButtons[0]);

    const newState = store.getState();

    expect(newState.alertReducer.open).toBeFalsy();
  });

  describe("getIcon()", () => {
    it("Should return the correct icon if the type passed is success", () => {
      const type = "success";
      const response = getIcon(type);

      expect(response).toEqual(<CheckCircle />);
    });
    it("Should return the correct icon if the type passed is not success", () => {
      const type = "error";
      const response = getIcon(type);

      expect(response).toEqual(<CircleXmark />);
    });
  });
});
