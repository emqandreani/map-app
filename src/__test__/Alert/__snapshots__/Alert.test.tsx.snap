// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Should render without crash 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedSuccess StyleSystem__Alert_outlinedSuccess MuiAlert-outlined css-zpxbnq-MuiPaper-root-MuiAlert-root"
  data-testid="alert"
  data-type="false"
  role="alert"
  style="background-color: rgb(255, 255, 255);"
>
  <div
    class="MuiAlert-icon StyleSystem__Alert_icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class=""
      height="16"
      unicode="f00c"
      viewBox="0 0 448 512"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"
        fill="var(--color-text-base)"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message StyleSystem__Alert_message css-1pxa9xg-MuiAlert-message"
  >
    <div
      class="MuiGrid-root MuiGrid-container StyleSystem__Alert_container css-1dphqhe-MuiGrid-root"
    >
      <div
        class="MuiGrid-root StyleSystem__Alert_text css-1gsico8-MuiGrid-root"
      >
        Colaborador creado con Ã©xito!
      </div>
    </div>
  </div>
  <div
    class="MuiAlert-action StyleSystem__Alert_action css-ki1hdl-MuiAlert-action"
  >
    <button
      aria-label="Close"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      title="Close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-1r0r0n5-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
